cpp/src/groupby/sort/sort_helper.cu

```diff
@@ -191,16 +145,17 @@ sort_groupby_helper::index_vector const& sort_groupby_helper::group_offsets(
 
   _group_offsets = std::make_unique<index_vector>(num_keys(stream) + 1, stream);
 
-  auto device_input_table = table_device_view::create(_keys, stream);
-  auto sorted_order       = key_sort_order(stream).data<size_type>();
+  auto const comparator  = cudf::experimental::row::equality::self_comparator{_keys, stream};
+  auto const d_key_equal = comparator.equal_to(
+    cudf::nullate::DYNAMIC{cudf::has_nested_nulls(_keys)}, null_equality::EQUAL);
+  auto const sorted_order = key_sort_order(stream).data<size_type>();
   decltype(_group_offsets->begin()) result_end;
 
-  result_end = thrust::unique_copy(
-    rmm::exec_policy(stream),
-    thrust::make_counting_iterator<size_type>(0),
-    thrust::make_counting_iterator<size_type>(num_keys(stream)),
-    _group_offsets->begin(),
-    permuted_row_equality_comparator(*device_input_table, sorted_order, has_nulls(_keys)));
+  result_end = thrust::unique_copy(rmm::exec_policy(stream),
+                                   thrust::make_counting_iterator<size_type>(0),
```

Comment:
The iterator template type is explicitly specified here. Probably should be consistent in both places. I think `size_type == int` is effectively the default that the compiler will infer anyway, so this is mostly for consistency. I don't think it will affect correctness, but let me know if I'm wrong.

Reply:
If you use `thrust::make_` then don't specify a template type. Otherwise, use `thrust::counting_iterator<type>()`.

Reply:
Using `thrust::counting_iterator<size_type>` consistently throughout the PR.